### ISSUE TITLE
Can't use threaded mode

### ISSUE THREAD

**Issue by marshygeek** [eternnoir/pyTelegramBotAPI#409]
1. What version of pyTelegramBotAPI are you using?
 Latest version of pyTelegramBotAPI
2. What OS are you using?
Ubuntu 16.04
3. What version of python are you using?
 python 3.6

My bot can't execute process_new_updates method, if there is no "threaded" flag (bot = telebot.TeleBot(config.TOKEN, threaded=False)). Is there is some better way to fix this?
I'm running my bot via nginx+flask.
I found that people encountered this problem already ([#155](https://github.com/eternnoir/pyTelegramBotAPI/issues/155) , [#132](https://github.com/eternnoir/pyTelegramBotAPI/issues/132)).

**Comment by 0xVK** [comment]
Do you use pythonanywhere for host?

**Comment by marshygeek** [comment]
No, I'm using AWS EC2.

**Comment by 0xVK** [comment]
I had the similar problem on pythonanywhere. And the reason was that host allowed run scripts only in one thread.

**Comment by marshygeek** [comment]
Thanks for reply, I'll check that

**Comment by marshygeek** [comment]
Thanks, you brought me to solution.
The problem was that uwsgi, by default, doesn't allow python plugin to initialize the GIL, so this means  that your app-generated threads will not run.
Problem was solved by editing configuration file uwsgi.ini:
Before:
`[uwsgi]
module = application:application

master = true
processes = 5

socket = tradebot.sock
chmod-socket = 666

vacuum = true

die-on-term = true`
After:
`[uwsgi]
module = application:application

processes = 1
enable-threads = true
single-interpreter = true

socket = tradebot.sock
chmod-socket = 666

vacuum = true

die-on-term = true`
The flag "master" causing trouble too. I found explanation [there](https://docs.newrelic.com/docs/agents/python-agent/hosting-mechanisms/python-agent-uwsgi)
